test/test_integration_credentials_netrc.py

```diff
@@ -0,0 +1,413 @@
+        utils.setup_cli(self)
+        with patch('getpass.getpass', return_value=self.client_params.env_api_key):
+            output = self.invoke_cli(self.cli_auth_params, ['login'])
+
+            assert utils.get_credentials() is not None
+            self.assertIn("Successfully logged in to Conjur", output.strip())
+            self.validate_netrc(f"{self.client_params.hostname}", "admin", self.client_params.env_api_key)
+
+    '''
+        Validates login is successful for hosts
+
+        Note we need to create the host first and rotate it's API key so that we can access it.
+        There is currently no way to fetch a host's API key so this is a work around for the 
+        purposes of this test
+        '''
+
+    @integration_test()
+    def test_https_netrc_is_created_with_host(self,mocks):
+        utils.setup_cli(self)
+        # Setup for fetching the API key of a host. To fetch we need to login
+        credentials = CredentialsData(self.client_params.hostname, "admin", self.client_params.env_api_key)
+        utils.save_credentials(credentials)
+
+        self.invoke_cli(self.cli_auth_params,
+                        ['policy', 'replace', '-b', 'root', '-f',
+                         self.environment.path_provider.get_policy_path('conjur')])
+
+        host_api_key = self.invoke_cli(self.cli_auth_params,
+                                       ['host', 'rotate-api-key', '-i', 'somehost'])
+
+        extract_api_key_from_message = host_api_key.split(":")[1].strip()
+
+        self.invoke_cli(self.cli_auth_params,
+                        ['logout'])
+
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['login', '-i', 'host/somehost', '-p', f'{extract_api_key_from_message}'])
+
+        self.validate_netrc(f"{self.client_params.hostname}", "host/somehost", extract_api_key_from_message)
+        self.assertIn("Successfully logged in to Conjur", output.strip())
+
+    @integration_test()
+    def test_working_with_FileProvider(self,mocks):
+        utils.setup_cli(self)
+        cred_store, _ = CredentialStoreFactory().create_credential_store()
+        self.assertEquals(type(cred_store), type(FileCredentialsProvider()))
+
+    '''
+        Validates logout doesn't remove another entry not associated with Cyberark
+        '''
+
+    @integration_test(True)
+    def test_https_netrc_does_not_remove_irrelevant_entry(self,mocks):
+        utils.setup_cli(self)
+        creds = CredentialsData(self.client_params.hostname, "admin", self.client_params.env_api_key)
+        utils.save_credentials(creds)
+
+        creds = CredentialsData("somemachine", "somelogin", "somepass")
+        utils.save_credentials(creds)
+
+        self.invoke_cli(self.cli_auth_params,
+                        ['logout'])
+        cred_store, _ = CredentialStoreFactory().create_credential_store()
+        assert cred_store.is_exists("somemachine")
+        assert not cred_store.is_exists(self.client_params.hostname)
```

Comment:
why shouldnt this exist? You have two different credentials that are not overriding each other

Reply:
we logged out